### PR TITLE
[AIR] Enable other notebooks previously marked with # REGRESSION

### PR DESCRIPTION
**Pull request by jiaodong**
## Why are these changes needed?

Fix notebooks and enable the ones that pass on CI

## Related issue number

#26410

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/data/examples/BUILD [hunk 1]
```diff
@@ -14,13 +14,12 @@ filegroup(
...
-    size = "medium",
+    size = "large",
     include = ["*.ipynb"],
     exclude = [
         "big_data_ingestion.ipynb",
-        "nyc_taxi_basic_processing.ipynb", # REGRESSION
     ],
     data = ["//doc/source/data/examples:data_examples"],
     tags = ["exclusive", "team:ml"],
-    allow_empty = True  # REGRESSION
```
Comment (richardliaw) [inline doc/source/data/examples/BUILD:25]: i think you can just remove this line

**Comment by jiaodong** [comment]
@richardliaw so besides the bucket reading issue on commit that i'm fixing now, the taxi dataset notebook runs e2e on large instance but will run out of disk space + timeout on CI machines. With repro-ci that's what i've been consistently seeing.

I think we should either trim down the size of this notebook or run it on nightly release tests instead. It's too heavy for CI.

**Comment by richardliaw** [comment]
Would it be helpful to ask others like @clarkzinzow to take a look?

**Comment by jiaodong** [comment]
Chatted with Matt a bit and i will update this PR with all fixes needed on other notebooks and separate the data folder's nyc taxi as separate task. We likely just need to move it as release test

**Review by xwjiang2010 — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]

**Pull request merged**